### PR TITLE
workaround for failing dask zeros_like

### LINKED ISSUE #4502: dask dev upstream test failiures
Our upstream tests fail due to a change in dask. The likely culprit is dask/dask#6680 - `dask.array.zeros_like` does now take the `meta` keyword and thus it tries to coerce `str` to a `bool` which fails (and it didn't do before). The following should trigger the error:

```python
import xarray as xr
import dask

da = xr.DataArray(dask.array.array(""))
xr.zeros_like(da, dtype=bool)
```
Note that `zeros_like` is called in `isnull` (which triggered the test failures):

https://github.com/pydata/xarray/blob/080caf4246fe2f4d6aa0c5dcb65a99b376fa669b/xarray/core/duck_array_ops.py#L100-L102

**What happened**:

```python
/home/vsts/work/1/s/xarray/tests/test_duck_array_ops.py:499: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/vsts/work/1/s/xarray/testing.py:139: in compat_variable
    return a.dims == b.dims and (a._data is b._data or equiv(a.data, b.data))
/home/vsts/work/1/s/xarray/testing.py:36: in _data_allclose_or_equiv
    return duck_array_ops.array_equiv(arr1, arr2)
/home/vsts/work/1/s/xarray/core/duck_array_ops.py:246: in array_equiv
    flag_array = (arr1 == arr2) | (isnull(arr1) & isnull(arr2))
/home/vsts/work/1/s/xarray/core/duck_array_ops.py:104: in isnull
    return zeros_like(data, dtype=bool)
/home/vsts/work/1/s/xarray/core/duck_array_ops.py:56: in f
    return wrapped(*args, **kwargs)
/usr/share/miniconda/envs/xarray-tests/lib/python3.8/site-packages/dask/array/creation.py:174: in zeros_like
    return zeros(
/usr/share/miniconda/envs/xarray-tests/lib/python3.8/site-packages/dask/array/wrap.py:78: in wrap_func_shape_as_first_arg
    return Array(dsk, name, chunks, dtype=dtype, meta=kwargs.get("meta", None))
/usr/share/miniconda/envs/xarray-tests/lib/python3.8/site-packages/dask/array/core.py:1083: in __new__
    meta = meta_from_array(meta, dtype=dtype)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

x = array('', dtype='<U1'), ndim = 0, dtype = dtype('bool')

    def meta_from_array(x, ndim=None, dtype=None):
        """Normalize an array to appropriate meta object
    
        Parameters
        ----------
        x: array-like, callable
            Either an object that looks sufficiently like a Numpy array,
            or a callable that accepts shape and dtype keywords
    
        if dtype and meta.dtype != dtype:
>           meta = meta.astype(dtype)
E           ValueError: invalid literal for int() with base 10: ''

```

### PR DESCRIPTION
**Pull request by mathause**
- [x] Closes #4502
 - [x] Passes `isort . && black . && mypy . && flake8`
 - [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/core/duck_array_ops.py [hunk 1]
```diff
@@ -101,6 +101,8 @@ def isnull(data):
         return isnan(data)
     elif issubclass(scalar_type, (np.bool_, np.integer, np.character, np.void)):
         # these types cannot represent missing values
+        if hasattr(data, "_meta"):
```
Comment (dcherian) [inline xarray/core/duck_array_ops.py:104]: We have `is_duck_dask_array` somewhere. But I'm not sure if `pint` (for e.g.) forwards the `_meta` property. cc @keewis

Comment (keewis) [reply xarray/core/duck_array_ops.py:104]: `pint` will forward to the wrapped array using `__getattr__` if it doesn't define the attribute itself, so we should be good. I'm not sure if other duck dask arrays do that, though (are there any rules/conventions for this?).

**Comment by mathause** [comment]
This might get fixed upstream dask/dask#6731 Also my idea does not seem to work...

**Comment by mathause** [comment]
This was fixed upstream. There are new failures but I am pretty sure they are unrelated as dask is not involved.

**Pull request closed without merging**